src/callbacks_step/analysis_surface_integral_2d.jl

```diff
@@ -103,28 +117,140 @@ which stores the boundary information and semidiscretization.
+- `uinf::Real`: Free-stream velocity
+- `linf::Real`: Reference length of geometry (e.g. airfoil chord length)
 """
-function DragCoefficientPressure(aoa, rhoinf, uinf, l_inf)
+function DragCoefficientShearStress(aoa, rhoinf, uinf, linf)
     # `psi_drag` is the unit vector tangent to the freestream direction
     psi_drag = (cos(aoa), sin(aoa))
-    return DragCoefficientPressure(ForceState(psi_drag, rhoinf, uinf, l_inf))
+    return DragCoefficientShearStress(ForceState(psi_drag, rhoinf, uinf, linf))
 end
 
 function (lift_coefficient::LiftCoefficientPressure)(u, normal_direction, equations)
     p = pressure(u, equations)
-    @unpack psi, rhoinf, uinf, l_inf = lift_coefficient.force_state
+    @unpack psi, rhoinf, uinf, linf = lift_coefficient.force_state
     # Normalize as `normal_direction` is not necessarily a unit vector
     n = dot(normal_direction, psi) / norm(normal_direction)
-    return p * n / (0.5 * rhoinf * uinf^2 * l_inf)
+    return p * n / (0.5 * rhoinf * uinf^2 * linf)
 end
 
 function (drag_coefficient::DragCoefficientPressure)(u, normal_direction, equations)
     p = pressure(u, equations)
-    @unpack psi, rhoinf, uinf, l_inf = drag_coefficient.force_state
+    @unpack psi, rhoinf, uinf, linf = drag_coefficient.force_state
     # Normalize as `normal_direction` is not necessarily a unit vector
     n = dot(normal_direction, psi) / norm(normal_direction)
-    return p * n / (0.5 * rhoinf * uinf^2 * l_inf)
+    return p * n / (0.5 * rhoinf * uinf^2 * linf)
+end
+
+# Compute the three different components of the viscous stress tensor
+# (tau_11, tau_12, tau_22) based on the gradients of the velocity field.
+# This is required for drag and lift coefficients based on shear stress, 
+# as well as for the non-integrated quantities such as 
+# skin friction coefficient (to be added).
+function viscous_stress_tensor(u, normal_direction, equations_parabolic,
+                               gradients_1, gradients_2)
+    _, dv1dx, dv2dx, _ = convert_derivative_to_primitive(u, gradients_1,
+                                                         equations_parabolic)
+    _, dv1dy, dv2dy, _ = convert_derivative_to_primitive(u, gradients_2,
+                                                         equations_parabolic)
+
+    # Components of viscous stress tensor
+    # (4/3 * (v1)_x - 2/3 * (v2)_y)
+    tau_11 = 4.0 / 3.0 * dv1dx - 2.0 / 3.0 * dv2dy
+    # ((v1)_y + (v2)_x)
+    # stress tensor is symmetric
+    tau_12 = dv1dy + dv2dx # = tau_21
+    # (4/3 * (v2)_y - 2/3 * (v1)_x)
+    tau_22 = 4.0 / 3.0 * dv2dy - 2.0 / 3.0 * dv1dx
+
+    mu = dynamic_viscosity(u, equations_parabolic)
+
+    return mu .* (tau_11, tau_12, tau_22)
+end
+
+function viscous_stress_vector(u, normal_direction, equations_parabolic,
+                               gradients_1, gradients_2)
+    #  Normalize normal direction, should point *into* the fluid => *(-1)
+    n_normal = -normal_direction / norm(normal_direction)
```

Comment:
In the paper from which we match the reference value, this normal is defined as " $\boldsymbol n$ is the outward normal at domain boundary". Doesn't it mean that it is pointing out (into the structure)?

Reply:
Yes, but this can't be right - remember the classic 1D case where the y-axis points into the fluid, see e.g. https://www.homepages.ucl.ac.uk/~uceseug/Fluids2/Notes_Viscosity.pdf

I think they just dropped the minus or used the values according to convention later on.